analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -220,15 +220,20 @@ class GoMod(
 
         return if (version.isBlank()) {
             // If the version is blank, it is a project in ORT speak.
-            check(main) { "Found a local module dependency which is not supported." }
-
             checkNotNull(dir) { "For projects, the directory is expected to not be null." }
 
+            val projectDir = File(dir)
+
+            require(projectDir.absoluteFile.startsWith(analysisRoot.absoluteFile)) {
+                "A replace directive references a module in '${projectDir.absolutePath}' outside of analysis root " +
```

Comment:
How about using `val projectDir = File(dir).absoluteFile` in line 225 so you can just use `$projectDir` here, and also simplify line 227?